src/trajectory.jl

```diff
@@ -159,11 +138,86 @@ function transition(rng::AbstractRNG, nt::NoUTurnTrajectory{I}, h::Hamiltonian,
         j = j + 1
     end
 
-    return θ_new, r_new, α / nα
+    H_new = 0 # Warning: NUTS always return H_new = 0;
```

Comment:
I guess after we remove the `step` interface, we need to compute `H_new` here. Otherwise the divergence check (`EBFMI`, which based on the list of `H`) in `sample()` would be wrong.

Reply:
yes, this is temporary. I plan to introduce a `PhasePoint` type to cache Hamiltonian energy.